### PR TITLE
MB-8240 Move hint text to live below input labels

### PR DESCRIPTION
**Pull request by Shkeating**
## Description

This branch moves hint text to live in between their respective input labels and text fields instead of being at the bottom. This will make the content more accessible to folks using assistive technology such as a screen reader. We want there users to receive the important information in the hint text before they potentially type in avoidable mistakes into a text field.

This branch also switches all hint text wherever possible to use the MMDS `Hint` component.

## Reviewer Notes
I tried my best to find all the forms in the repo that use hint text and relocate them to the correct location. This change also has not been completely reflected in all designs yet. Moving forward this will be the expected implementation. Please try your best to remember this when doing frontend work and when reviewing the work of others.

## Setup

```sh
yarn storybook
```

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8240?atlOrigin=eyJpIjoiNWUwNDg5MTQ1ZDY2NDc3MzkyMTM3NmQ4YTE2ZDNiZDEiLCJwIjoiaiJ9) for this change
* [Zeplin reference](https://zpl.io/V1xvLn4) 

## Screenshots
[coming soon!]

### CODE REVIEW

File: src/components/Office/ServicesCounselingShipmentForm/ServicesCounselingShipmentForm.jsx [hunk 1]
```diff
@@ -347,6 +347,9 @@ const ServicesCounselingShipmentForm = ({
                       Remarks <span className="float-right">{optionalLabel}</span>
                     </h2>
                     <Label htmlFor="customerRemarks">Customer remarks</Label>
+                    <Hint>
```
Comment (YanZ777) [inline src/components/Office/ServicesCounselingShipmentForm/ServicesCounselingShipmentForm.jsx:350]: There's a customer remarks field in the MtoShipment form as well. Should the 250 character hint be moved up on that one as well?

**Comment by YanZ777** [comment]
Hint text that I've seen that hasn't been moved yet:
- `OrdersInfoForm`: "Date your orders were issued."
- `UploadOrders`: "(Each page must be clear and legible.)"

**Review by YanZ777 — COMMENTED** [review]
This can probably be left to another ticket, but I did notice that the hint text is now very far away from the field on the allowances detail form.

**Comment by Shkeating** [comment]
> This can probably be left to another ticket, but I did notice that the hint text is now very far away from the field on the allowances detail form.

I sorted it out because it was easy!

**Review by YanZ777 — APPROVED** [review]
Thanks for fixing the spacing! Looks good to me!

**Comment by cosburn** [comment]
Amazing work! Everything looks good, I did notice one component in Happo with what looks like hint text below the field. Is this component something we want to update, or are we focusing on the customer app?

Screen Shot 2021-07-20 at 9 34 21 AM

**Comment by Shkeating** [comment]
> Amazing work! Everything looks good, I did notice one component in Happo with what looks like hint text below the field. Is this component something we want to update, or are we focusing on the customer app?
> 
> Screen Shot 2021-07-20 at 9 34 21 AM

In this component, it can't be moved due to how the `Masked Text Field` component that it uses works :/ It felt out of scope to do the level of refactor needed in there to sort it out.

**Review by cosburn — APPROVED** [review]
Nice!

**Pull request merged**